base/inference.jl

```diff
@@ -1269,7 +1269,6 @@ function tmerge(typea::ANY, typeb::ANY)
         if isa(typea, DataType) && isa(typeb, DataType) && length(typea.parameters) == length(typeb.parameters) && !isvatuple(typea) && !isvatuple(typeb)
             return typejoin(typea, typeb)
         end
-        return Tuple
```

Comment:
Not sure. We need to do something here, so that it's possible to infer the result of `(tup::Union{Tuple{Bool}, Tuple{Bool,Int}})[1]`. Maybe allow a max of only 2 tuple types in a union?